ui/helpers/utils/util.js

```diff
@@ -87,16 +87,14 @@ export function addressSummary(
     : '...';
 }
 
-export function isValidDomainName(address) {
+export function isValidEnsDomainName(address) {
   const match = punycode
     .toASCII(address)
     .toLowerCase()
     // Checks that the domain consists of at least one valid domain pieces separated by periods, followed by a tld
     // Each piece of domain name has only the characters a-z, 0-9, and a hyphen (but not at the start or end of chunk)
     // A chunk has minimum length of 1, but minimum tld is set to 2 for now (no 1-character tlds exist yet)
-    .match(
-      /^(?:[a-z0-9](?:[-a-z0-9]*[a-z0-9])?\.)+[a-z0-9][-a-z0-9]*[a-z0-9]$/u,
-    );
+    .match(/^(?:[a-z0-9](?:[-a-z0-9]*[a-z0-9])?\.)+eth$/u);
```

Comment:
This constitutes a change in logic, disallowing previous TLDS that would resovle properly by ENS. This won't work, we'll have to find a solution that will allow both ENS and UD to resolve these TLDS based on user preference.